### PR TITLE
:lipstick: feat: Create reusable news item component #11

### PR DESCRIPTION
**Pull request by Darylcosm0**
# Description

Please include a summary of the changes and the related issue. Please also include relevant motivation and context. List any dependencies that are required for this change.

Reference the issue related to this PR as shown below. Every issue has it's own number you can find the issue numbers [here](https://github.com/italanta/italanta-apps/issues).

Fixes # 11

## Type of change

Please delete options that are not relevant.

- [x] New feature (non-breaking change which adds functionality)

## Screenshot (optional)
Screenshot from 2023-08-16 14-47-13

## How Has This Been Tested?

Please describe the tests that you ran to verify your changes. Provide instructions so we can reproduce. Please also list any relevant details for your test configuration

**Test Configuration**: (optional)

- Firmware version:
- Hardware:
- Toolchain:
- SDK:

## Checklist

- [x] My code follows the style guidelines of this project
- [x] I have performed a self-review of my code
- [x] I have commented my code, particularly in hard-to-understand areas
- [x] My changes generate no new warnings- [ ] New and existing unit tests pass locally with my changes

### CODE REVIEW

File: apps/elewa-website/src/app/app.routing.ts [hunk 1]
```diff
@@ -10,8 +10,13 @@ export const ELEWA_WEBSITE_ROUTES: Route[] = [
   { path: 'home', loadChildren: () => import('@elewa-website/features/pages/home').then(m => m.FeaturesPagesHomeModule) },
 
   { path: 'about', loadChildren: () => import('@elewa-website/features/pages/about').then(m => m.FeaturesPagesAboutModule) },
+
+  { path: 'cards', loadChildren: () => import('@elewa-website/elements/cards').then(m => m.CardsModule) },
+
```
Comment (LemmyMwaura) [inline apps/elewa-website/src/app/app.routing.ts:15]: No need for the route as this is not a page

File: libs/elements/cards/src/lib/cards.routing.ts [hunk 2]
```diff
@@ -0,0 +1,14 @@
+import { NgModule } from '@angular/core';
+import { RouterModule, Routes } from '@angular/router';
+import { ElewaNewsItemCardComponent } from './components/elewa-news-item-card/elewa-news-item-card.component';
+
+export const CARDS_ROUTES: Routes = [
+  { path: '', component: ElewaNewsItemCardComponent },
+
+];
+
```
Comment (LemmyMwaura) [inline libs/elements/cards/src/lib/cards.routing.ts:9]: No need for the routing

File: libs/elements/cards/src/lib/components/elewa-news-item-card/elewa-news-item-card.component.html [hunk 3]
```diff
@@ -0,0 +1,14 @@
+<div *ngFor="let item of newsItem" class="news-item">
+  <!-- title -->
+  <div class="title">
+    {{ item.title }}
```
Comment (LemmyMwaura) [inline libs/elements/cards/src/lib/components/elewa-news-item-card/elewa-news-item-card.component.html:4]: This component should only render one item. No need for the loop

File: libs/elements/cards/src/lib/components/elewa-news-item-card/elewa-news-item-card.component.ts [hunk 4]
```diff
@@ -0,0 +1,17 @@
...
+  selector: 'elewa-website-elewa-news-item-card',
+  templateUrl: './elewa-news-item-card.component.html',
+  styleUrls: ['./elewa-news-item-card.component.scss'],
+})
+export class ElewaNewsItemCardComponent implements OnInit  {
+  @Input() newsItem!: NewsItem[]
+  
+  ngOnInit() {
+      this.newsItem = highlightedNewsData
+  }
```
Comment (LemmyMwaura) [inline libs/elements/cards/src/lib/components/elewa-news-item-card/elewa-news-item-card.component.ts:16]: Should only render one item. The loop will happen on the parent component to this, which will be created in another issue.

**Review by LemmyMwaura — CHANGES_REQUESTED** [review]
Just a few changes left. Also clean up the design to match the spec to the best of your ability

File: libs/elements/cards/src/lib/components/elewa-news-item-card/elewa-news-item-card.component.html (hunk 3 above)
Comment (Darylcosm0) [reply libs/elements/cards/src/lib/components/elewa-news-item-card/elewa-news-item-card.component.html:4]: i have fixed [<div class="news-item">
  
  <div class="title">
    {{ newsItem[0].title }}
  </div>
  
  <div class="description">
    {{ newsItem[0].description }}
  </div>
  
  <div class="button-text">
    {{ newsItem[0].buttonText }}
  </div>
</div>
]

File: apps/elewa-website/src/app/app.routing.ts (hunk 1 above)
Comment (LemmyMwaura) [reply apps/elewa-website/src/app/app.routing.ts:15]: For me to merge this, Yes as it's redundant.

File: libs/elements/cards/src/lib/cards.routing.ts (hunk 2 above)
Comment (Darylcosm0) [reply libs/elements/cards/src/lib/cards.routing.ts:9]: removed

File: apps/elewa-website/src/app/app.routing.ts (hunk 1 above)
Comment (Darylcosm0) [reply apps/elewa-website/src/app/app.routing.ts:15]: okay solved

File: libs/elements/layout/text/src/lib/text.routing.ts [hunk 5]
```diff
@@ -0,0 +1,15 @@
+import { NgModule } from '@angular/core';
```
Comment (LemmyMwaura) [inline libs/elements/layout/text/src/lib/text.routing.ts:1]: There's no need for routing on this file.

File: libs/models/elements/layout/images/src/lib/models-elements-layout-images.module.ts [hunk 6]
```diff
@@ -0,0 +1,7 @@
+import { NgModule } from '@angular/core';
+import { CommonModule } from '@angular/common';
+
+@NgModule({
+  imports: [CommonModule],
+})
+export class ModelsElementsLayoutImagesModule {}
```
Comment (LemmyMwaura) [inline libs/models/elements/layout/images/src/lib/models-elements-layout-images.module.ts:7]: rename this to just ImagesModelsModule if the name was not provided

Comment (LemmyMwaura) [reply libs/models/elements/layout/images/src/lib/models-elements-layout-images.module.ts:7]: Nevermind. This is a completely different issue. The changes shouldn't be part of this branch

**Review by LemmyMwaura — CHANGES_REQUESTED** [review]

**Comment by LemmyMwaura** [comment]
Closing this as it's been merged in #87

**Pull request closed without merging**